source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -104,6 +104,191 @@ TEST(XnetConvertersTests, SessionInfoStateValue_SetSessionInfoResponse_ExtractSe
   EXPECT_EQ(nixnet_grpc::SessionInfoState::SESSION_INFO_STATE_STARTED, output.info());
   EXPECT_EQ(sessionInfoState, output.info_raw());
 }
+
+void assert_frames_are_equal(nxFrameVar_t* frame1, nixnet_grpc::FrameResponse frame2)
```

Comment:
Move all helper methods and struct definition at top of file before all the tests begin. You can keep the following structure.

> struct EthernetHeader ...
> Ethernet frame helpers ...
> assert helper methods ...
> all tests ...

First two above will be moved from here to `nixnet_utilities.h` by Akshata. We figured out what the linking error was and she will be fixing it in her system tests PR.

Reply:
Moved all helper functions to nixnet_utilities.h header file and assert helper functions at the top of file.